test/scripts/txsender/main.go

```diff
@@ -0,0 +1,210 @@
+}
+
+var (
+	flagNetwork = cli.StringSliceFlag{
+		Name:     flagNetworkName,
+		Aliases:  []string{"n"},
+		Usage:    "list of networks on which to send transactions",
+		Required: false,
+	}
+	flagWait = cli.BoolFlag{
+		Name:     flagWaitName,
+		Aliases:  []string{"w"},
+		Usage:    "wait transactions to be confirmed",
+		Required: false,
+	}
+	flagVerbose = cli.BoolFlag{
+		Name:     flagVerboseName,
+		Aliases:  []string{"v"},
+		Usage:    "output verbose logs",
+		Required: false,
+	}
+)
+
+func main() {
+	txsender := cli.NewApp()
+	txsender.Name = "txsender"
+	txsender.Flags = []cli.Flag{&flagNetwork, &flagWait, &flagVerbose}
+	txsender.Usage = "send transactions"
+	txsender.Description = `This tool allows to send a specified number of transactions.
+Optionally it can wait for the transactions to be validated.`
+	txsender.DefaultCommand = "send"
+	txsender.Commands = []*cli.Command{
+		{
+			Name:    "send",
+			Aliases: []string{},
+			Usage:   "Sends the specified number of transactions",
+			Description: `This command sends the specified number of transactions.
+If --wait flag is used, it waits for the corresponding validation transaction.`,
+			ArgsUsage: "number of transactions to be sent (default: 1)",
+			Action:    sendTxs,
+		},
+	}
+
+	err := txsender.Run(os.Args)
+	if err != nil {
+		log.Fatal(err)
+		os.Exit(1)
+	}
+}
+
+func sendTxs(cliCtx *cli.Context) error {
+	ctx := cliCtx.Context
+
+	logLevel := "info"
+	if cliCtx.Bool(flagVerboseName) {
+		logLevel = "debug"
+	}
+
+	log.Init(log.Config{Level: logLevel, Outputs: []string{"stdout"}})
+
+	nTxs := 1 // send 1 tx by default
+	if cliCtx.NArg() > 0 {
+		nTxsArgStr := cliCtx.Args().Get(0)
+		nTxsArg, err := strconv.Atoi(nTxsArgStr)
+		if err == nil {
+			nTxs = nTxsArg
+		}
+	}
+
+	selNetworks := cliCtx.StringSlice(flagNetworkName)
+
+	// if no network selected, pick the default one
+	if selNetworks == nil {
+		selNetworks = []string{defaultNetwork}
+	}
+
+	for _, selNetwork := range selNetworks {
+		network, ok := networks[selNetwork]
+		if !ok {
+			netKeys := make([]string, 0, len(networks))
+			for net := range networks {
+				netKeys = append(netKeys, net)
+			}
+			return fmt.Errorf("Please specify one or more of these networks: %v", netKeys)
+		}
+		log.Infof("connecting to %v: %v", network.Name, network.URL)
+		client, err := ethclient.Dial(network.URL)
+		if err != nil {
+			return err
+		}
+		log.Infof("connected")
+
+		auth, err := operations.GetAuth(network.PrivateKey, network.ChainID)
+		if err != nil {
+			return err
+		}
+
+		log.Infof("Sender Addr: %v", auth.From.String())
+
+		senderBalance, err := client.BalanceAt(ctx, auth.From, nil)
+		if err != nil {
+			return err
+		}
+		log.Debugf("ETH Balance for %v: %v", auth.From, senderBalance)
+
+		amount := big.NewInt(10) //nolint:gomnd
+		log.Debugf("Transfer Amount: %v", amount)
+
+		senderNonce, err := client.PendingNonceAt(ctx, auth.From)
```

Comment:
I would like to customize the starting nonce and the gasPrice. Could you add this tiny improvement? This will allow us to unblock transactions due to nonce or gasPrice problems if it happens. @ToniRamirezM what do you think?

Reply:
Yes, this could be usefull in some cases. I don't know how good idea is to put it here of leave that for another espeficic script.

Reply:
@ARR552 I resolve this in order to merge the PR. We can add that feature in a second PR if you agree.
I was thinking about adding a `--file` flag to parse transactions from a json file, WDYT?